xtask/src/ci.rs

```diff
@@ -102,25 +102,16 @@ fn e2e_test(sh: &Shell, args: Vec<OsString>) -> anyhow::Result<()> {
     let err_context = "failed to run E2E tests";
 
     let _pd = sh.push_dir(crate::project_root());
-    e2e_build(sh).context(err_context)?;
-
-    // Build debug plugins for test binary
-    build::build(
-        sh,
-        flags::Build {
-            release: false,
-            no_plugins: false,
-            plugins_only: true,
-        },
-    )
-    .context(err_context)?;
 
     crate::cargo()
         .and_then(|cargo| {
-            cmd!(sh, "{cargo} test -- --ignored --nocapture --test-threads 1")
-                .args(args)
-                .run()
-                .map_err(anyhow::Error::new)
+            cmd!(
+                sh,
+                "{cargo} test --no-default-features -- --ignored --nocapture --test-threads 1"
+            )
+            .args(args)
+            .run()
+            .map_err(anyhow::Error::new)
```

Comment:
This makes zellij take whatever plugins lie in the `zellij-utils/assets/plugin/` folder. Why did you change this? Of course it speeds up the e2e tests, but we do not guarantee that the content of `zellij-utils/assets/plugins` is up to date and matches the current zellij version. That's only the case when building the plugins in release mode first, because then they're copied into the `assets/plugins` folder. Or does the e2e binary take the plugins from somewhere else entirely? Afaik it did a while ago, but iirc I changed this to make it bundle the plugins into the e2e binary.

Reply:
This had me confused for a while too. But as far as I understand it, we just need any plugins to be available so that the test binary can be built (because it depends on zellij_utils i think), so we might as well just use the already built ones.  And we already have them because we build the in `e2e_build`. 

The e2e test binary in the end is just running the release built zellij binary (built in `e2e_build`) through ssh machinery, and setting it up to use a custom 'e2e-data' dir so that it finds the release optimized plugins (built in `e2e_build`).

Probably I should change it to ensure that e2e_test always calls e2e_build, and have it as a single step in the github flow. That way we calling the command locally everything should always be up to date.

Hope the explanation makes sense, it's all a bit convoluted :)